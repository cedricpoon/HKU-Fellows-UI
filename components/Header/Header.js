import React, { Component } from 'react';
import PropTypes from "prop-types";
import { Left, Body, Right, Button, Icon, Title, Subtitle as NBSubtitle } from 'native-base';
import { Header as NBHeader } from 'hkufui/native-base-fork/Header';
import * as Animatable from 'react-native-animatable';

import NavigationService from 'hkufui/src/NavigationService';
import styles from './Styles';

const Subtitle = Animatable.createAnimatableComponent(NBSubtitle);

const subtitleAnimationDuration = 500;

class Header extends Component {
  constructor(props) {
    super(props);

    this.state = { subtitle: this.props.subtitle };
    this.setSubtitle = this.setSubtitle.bind(this);
  }

  setSubtitle(subtitle = null) {
    this.setState({
      subtitle: subtitle
    });
  }

  componentDidUpdate() {
    if (this._subtitle) {
      this._subtitle.pulse(subtitleAnimationDuration); // eslint-disable-line react/prop-types
    }
  }

  render() {
    const {
      title,
      backable,
      rightIcon,
      onRightPress,
      rightStyle,
      ...restProps
    } = this.props;

    const { subtitle } = this.state.subtitle ? this.state : this.props;

    return (
      <NBHeader {...restProps} style={styles.header}>
        <Left>
          { backable && (
            <Button
              transparent
              onPress={() => { NavigationService.goBack(); }}
            >
              <Icon style={styles.button} name="arrow-dropleft"/>
            </Button>
          )}
        </Left>
        <Body style={styles.context}>
          <Title
            style={[
              styles.title,
              { fontSize: title.size }
            ]}
            ref={ref => { this._title=ref }}
            numberOfLines={title.numberOfLines || 1}
          >
            {title.context}
          </Title>
          {subtitle && (
            <Subtitle
              ref={ref => { this._subtitle = ref }}
              style={styles.subtitle}
              numberOfLines={subtitle.numberOfLines || 1}
            >
              {subtitle.context}
            </Subtitle>
          )}
        </Body>
        <Right>
        {rightIcon && (
          <Button transparent onPress={onRightPress}>
            <Icon name={rightIcon} style={rightStyle} />
          </Button>
        )}
        </Right>
      </NBHeader>
    );
  }
}

Header.defaultProps = {
  onRightPress: () => {},
  subtitle: null,
  rightStyle: styles.button
};

const contextProptypes = {
  context: PropTypes.string.isRequired,
  numberOfLines: PropTypes.number,
  size: PropTypes.number
}

Header.propTypes = {
  title: PropTypes.shape(contextProptypes).isRequired,
  subtitle: PropTypes.shape(contextProptypes),
  backable: PropTypes.bool,
  rightIcon: PropTypes.string,
  rightStyle: Icon.propTypes.style,
  onRightPress: PropTypes.func
};

export default Header;
