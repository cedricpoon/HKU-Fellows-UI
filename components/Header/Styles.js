import { StyleSheet } from 'react-native';

import { fontSize, gridBase } from 'hkufui/theme/grid'
import { logo } from 'hkufui/theme/palette';

export default StyleSheet.create({
  header: {
    height: 'auto',
    paddingBottom: 0.5 // compensation on diff with PreviewHeader
  },
  title: {
    fontSize: fontSize.title,
    fontWeight: 'normal',
    textAlign: 'justify'
  },
  titleRegularSize: {
    fontSize: fontSize.titleSize
  },
  subtitle: {
    color: logo.brown,
    paddingTop: gridBase,
    textAlign: 'justify',
    marginBottom: gridBase
  },
  back: {
    color: logo.black
  },
  rightIcon: {
    color: logo.red
  },
  context: {
    flex: 4
  }
});
