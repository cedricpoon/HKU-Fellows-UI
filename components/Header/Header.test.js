import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Testing Header Component', () => {
  it('renders as expected with title', () => {
    const wrapper = shallow(
      <Header title="Header" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as expected with back button', () => {
    const wrapper = shallow(
      <Header title="Header" backable />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
