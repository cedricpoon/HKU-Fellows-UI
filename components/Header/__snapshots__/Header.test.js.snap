// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header Component renders as expected with back button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    backable={true}
    onRightPress={[Function]}
    subtitle={null}
    title="Header"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Left)
          style={
            Object {
              "flex": 0.2,
            }
          }
        >
          <Styled(Button)
            onPress={[Function]}
            transparent={true}
          >
            <Styled(Icon)
              name="arrow-dropleft"
              style={
                Object {
                  "color": "#4E4A47",
                }
              }
            />
          </Styled(Button)>
        </Styled(Left)>,
        <Styled(Body)>
          <Styled(Title)
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "normal",
              }
            }
          >
            Header
          </Styled(Title)>
        </Styled(Body)>,
        undefined,
      ],
      "subtitle": null,
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(Button)
            onPress={[Function]}
            transparent={true}
          >
            <Styled(Icon)
              name="arrow-dropleft"
              style={
                Object {
                  "color": "#4E4A47",
                }
              }
            />
          </Styled(Button)>,
          "style": Object {
            "flex": 0.2,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Icon)
              name="arrow-dropleft"
              style={
                Object {
                  "color": "#4E4A47",
                }
              }
            />,
            "onPress": [Function],
            "transparent": true,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "arrow-dropleft",
              "style": Object {
                "color": "#4E4A47",
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Title)
              style={
                Object {
                  "fontSize": 15,
                  "fontWeight": "normal",
                }
              }
            >
              Header
            </Styled(Title)>,
            null,
          ],
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Header",
              "style": Object {
                "fontSize": 15,
                "fontWeight": "normal",
              },
              "virtual": undefined,
            },
            "ref": [Function],
            "rendered": "Header",
            "type": [Function],
          },
          null,
        ],
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Left)
            style={
              Object {
                "flex": 0.2,
              }
            }
          >
            <Styled(Button)
              onPress={[Function]}
              transparent={true}
            >
              <Styled(Icon)
                name="arrow-dropleft"
                style={
                  Object {
                    "color": "#4E4A47",
                  }
                }
              />
            </Styled(Button)>
          </Styled(Left)>,
          <Styled(Body)>
            <Styled(Title)
              style={
                Object {
                  "fontSize": 15,
                  "fontWeight": "normal",
                }
              }
            >
              Header
            </Styled(Title)>
          </Styled(Body)>,
          undefined,
        ],
        "subtitle": null,
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Button)
              onPress={[Function]}
              transparent={true}
            >
              <Styled(Icon)
                name="arrow-dropleft"
                style={
                  Object {
                    "color": "#4E4A47",
                  }
                }
              />
            </Styled(Button)>,
            "style": Object {
              "flex": 0.2,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="arrow-dropleft"
                style={
                  Object {
                    "color": "#4E4A47",
                  }
                }
              />,
              "onPress": [Function],
              "transparent": true,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "arrow-dropleft",
                "style": Object {
                  "color": "#4E4A47",
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Title)
                style={
                  Object {
                    "fontSize": 15,
                    "fontWeight": "normal",
                  }
                }
              >
                Header
              </Styled(Title)>,
              null,
            ],
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Header",
                "style": Object {
                  "fontSize": 15,
                  "fontWeight": "normal",
                },
                "virtual": undefined,
              },
              "ref": [Function],
              "rendered": "Header",
              "type": [Function],
            },
            null,
          ],
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing Header Component renders as expected with right button 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    onRightPress={[Function]}
    rightIcon="arrow-dropup"
    subtitle={null}
    title="Header"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Left)
          style={
            Object {
              "flex": 0.2,
            }
          }
        />,
        <Styled(Body)>
          <Styled(Title)
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "normal",
              }
            }
          >
            Header
          </Styled(Title)>
        </Styled(Body)>,
        <Styled(Right)
          style={
            Object {
              "flex": 0.2,
            }
          }
        >
          <Styled(Button)
            onPress={[Function]}
            transparent={true}
          >
            <Styled(Icon)
              name="arrow-dropup"
              style={
                Object {
                  "color": "#E03A00",
                }
              }
            />
          </Styled(Button)>
        </Styled(Right)>,
      ],
      "subtitle": null,
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "style": Object {
            "flex": 0.2,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Title)
              style={
                Object {
                  "fontSize": 15,
                  "fontWeight": "normal",
                }
              }
            >
              Header
            </Styled(Title)>,
            null,
          ],
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Header",
              "style": Object {
                "fontSize": 15,
                "fontWeight": "normal",
              },
              "virtual": undefined,
            },
            "ref": [Function],
            "rendered": "Header",
            "type": [Function],
          },
          null,
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(Button)
            onPress={[Function]}
            transparent={true}
          >
            <Styled(Icon)
              name="arrow-dropup"
              style={
                Object {
                  "color": "#E03A00",
                }
              }
            />
          </Styled(Button)>,
          "style": Object {
            "flex": 0.2,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Icon)
              name="arrow-dropup"
              style={
                Object {
                  "color": "#E03A00",
                }
              }
            />,
            "onPress": [Function],
            "transparent": true,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "arrow-dropup",
              "style": Object {
                "color": "#E03A00",
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Left)
            style={
              Object {
                "flex": 0.2,
              }
            }
          />,
          <Styled(Body)>
            <Styled(Title)
              style={
                Object {
                  "fontSize": 15,
                  "fontWeight": "normal",
                }
              }
            >
              Header
            </Styled(Title)>
          </Styled(Body)>,
          <Styled(Right)
            style={
              Object {
                "flex": 0.2,
              }
            }
          >
            <Styled(Button)
              onPress={[Function]}
              transparent={true}
            >
              <Styled(Icon)
                name="arrow-dropup"
                style={
                  Object {
                    "color": "#E03A00",
                  }
                }
              />
            </Styled(Button)>
          </Styled(Right)>,
        ],
        "subtitle": null,
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "style": Object {
              "flex": 0.2,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Title)
                style={
                  Object {
                    "fontSize": 15,
                    "fontWeight": "normal",
                  }
                }
              >
                Header
              </Styled(Title)>,
              null,
            ],
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Header",
                "style": Object {
                  "fontSize": 15,
                  "fontWeight": "normal",
                },
                "virtual": undefined,
              },
              "ref": [Function],
              "rendered": "Header",
              "type": [Function],
            },
            null,
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Button)
              onPress={[Function]}
              transparent={true}
            >
              <Styled(Icon)
                name="arrow-dropup"
                style={
                  Object {
                    "color": "#E03A00",
                  }
                }
              />
            </Styled(Button)>,
            "style": Object {
              "flex": 0.2,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="arrow-dropup"
                style={
                  Object {
                    "color": "#E03A00",
                  }
                }
              />,
              "onPress": [Function],
              "transparent": true,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "arrow-dropup",
                "style": Object {
                  "color": "#E03A00",
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing Header Component renders as expected with subtitle 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    onRightPress={[Function]}
    subtitle="None"
    title="Header"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Left)
          style={
            Object {
              "flex": 0.2,
            }
          }
        />,
        <Styled(Body)>
          <Styled(Title)
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "normal",
              }
            }
          >
            Header
          </Styled(Title)>
          <withAnimatable(Styled(Subtitle))
            delay={0}
            direction="normal"
            iterationCount={1}
            iterationDelay={0}
            numberOfLines={1}
            onAnimationBegin={[Function]}
            onAnimationEnd={[Function]}
            onTransitionBegin={[Function]}
            onTransitionEnd={[Function]}
            style={
              Object {
                "color": "#685A5C",
                "paddingTop": 5,
              }
            }
            useNativeDriver={false}
          >
            None
          </withAnimatable(Styled(Subtitle))>
        </Styled(Body)>,
        undefined,
      ],
      "subtitle": "None",
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "style": Object {
            "flex": 0.2,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Title)
              style={
                Object {
                  "fontSize": 15,
                  "fontWeight": "normal",
                }
              }
            >
              Header
            </Styled(Title)>,
            <withAnimatable(Styled(Subtitle))
              delay={0}
              direction="normal"
              iterationCount={1}
              iterationDelay={0}
              numberOfLines={1}
              onAnimationBegin={[Function]}
              onAnimationEnd={[Function]}
              onTransitionBegin={[Function]}
              onTransitionEnd={[Function]}
              style={
                Object {
                  "color": "#685A5C",
                  "paddingTop": 5,
                }
              }
              useNativeDriver={false}
            >
              None
            </withAnimatable(Styled(Subtitle))>,
          ],
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Header",
              "style": Object {
                "fontSize": 15,
                "fontWeight": "normal",
              },
              "virtual": undefined,
            },
            "ref": [Function],
            "rendered": "Header",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "animation": undefined,
              "children": "None",
              "delay": 0,
              "direction": "normal",
              "duration": undefined,
              "easing": undefined,
              "iterationCount": 1,
              "iterationDelay": 0,
              "numberOfLines": 1,
              "onAnimationBegin": [Function],
              "onAnimationEnd": [Function],
              "onTransitionBegin": [Function],
              "onTransitionEnd": [Function],
              "style": Object {
                "color": "#685A5C",
                "paddingTop": 5,
              },
              "transition": undefined,
              "useNativeDriver": false,
            },
            "ref": [Function],
            "rendered": "None",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Left)
            style={
              Object {
                "flex": 0.2,
              }
            }
          />,
          <Styled(Body)>
            <Styled(Title)
              style={
                Object {
                  "fontSize": 15,
                  "fontWeight": "normal",
                }
              }
            >
              Header
            </Styled(Title)>
            <withAnimatable(Styled(Subtitle))
              delay={0}
              direction="normal"
              iterationCount={1}
              iterationDelay={0}
              numberOfLines={1}
              onAnimationBegin={[Function]}
              onAnimationEnd={[Function]}
              onTransitionBegin={[Function]}
              onTransitionEnd={[Function]}
              style={
                Object {
                  "color": "#685A5C",
                  "paddingTop": 5,
                }
              }
              useNativeDriver={false}
            >
              None
            </withAnimatable(Styled(Subtitle))>
          </Styled(Body)>,
          undefined,
        ],
        "subtitle": "None",
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "style": Object {
              "flex": 0.2,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Title)
                style={
                  Object {
                    "fontSize": 15,
                    "fontWeight": "normal",
                  }
                }
              >
                Header
              </Styled(Title)>,
              <withAnimatable(Styled(Subtitle))
                delay={0}
                direction="normal"
                iterationCount={1}
                iterationDelay={0}
                numberOfLines={1}
                onAnimationBegin={[Function]}
                onAnimationEnd={[Function]}
                onTransitionBegin={[Function]}
                onTransitionEnd={[Function]}
                style={
                  Object {
                    "color": "#685A5C",
                    "paddingTop": 5,
                  }
                }
                useNativeDriver={false}
              >
                None
              </withAnimatable(Styled(Subtitle))>,
            ],
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Header",
                "style": Object {
                  "fontSize": 15,
                  "fontWeight": "normal",
                },
                "virtual": undefined,
              },
              "ref": [Function],
              "rendered": "Header",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "animation": undefined,
                "children": "None",
                "delay": 0,
                "direction": "normal",
                "duration": undefined,
                "easing": undefined,
                "iterationCount": 1,
                "iterationDelay": 0,
                "numberOfLines": 1,
                "onAnimationBegin": [Function],
                "onAnimationEnd": [Function],
                "onTransitionBegin": [Function],
                "onTransitionEnd": [Function],
                "style": Object {
                  "color": "#685A5C",
                  "paddingTop": 5,
                },
                "transition": undefined,
                "useNativeDriver": false,
              },
              "ref": [Function],
              "rendered": "None",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing Header Component renders as expected with title 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    onRightPress={[Function]}
    subtitle={null}
    title="Header"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Left)
          style={
            Object {
              "flex": 0.2,
            }
          }
        />,
        <Styled(Body)>
          <Styled(Title)
            style={
              Object {
                "fontSize": 15,
                "fontWeight": "normal",
              }
            }
          >
            Header
          </Styled(Title)>
        </Styled(Body)>,
        undefined,
      ],
      "subtitle": null,
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "style": Object {
            "flex": 0.2,
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Title)
              style={
                Object {
                  "fontSize": 15,
                  "fontWeight": "normal",
                }
              }
            >
              Header
            </Styled(Title)>,
            null,
          ],
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Header",
              "style": Object {
                "fontSize": 15,
                "fontWeight": "normal",
              },
              "virtual": undefined,
            },
            "ref": [Function],
            "rendered": "Header",
            "type": [Function],
          },
          null,
        ],
        "type": [Function],
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Left)
            style={
              Object {
                "flex": 0.2,
              }
            }
          />,
          <Styled(Body)>
            <Styled(Title)
              style={
                Object {
                  "fontSize": 15,
                  "fontWeight": "normal",
                }
              }
            >
              Header
            </Styled(Title)>
          </Styled(Body)>,
          undefined,
        ],
        "subtitle": null,
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "style": Object {
              "flex": 0.2,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Title)
                style={
                  Object {
                    "fontSize": 15,
                    "fontWeight": "normal",
                  }
                }
              >
                Header
              </Styled(Title)>,
              null,
            ],
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Header",
                "style": Object {
                  "fontSize": 15,
                  "fontWeight": "normal",
                },
                "virtual": undefined,
              },
              "ref": [Function],
              "rendered": "Header",
              "type": [Function],
            },
            null,
          ],
          "type": [Function],
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
